reana_server/reana_admin/cli.py

```diff
@@ -716,3 +718,159 @@ def queue_consume(
         logging.exception(error)
     else:
         consumer.run()
+
+
+@reana_admin.command("check-workflows")
+@click.option(
+    "--date-start",
+    type=click.DateTime(formats=["%Y-%m-%d", "%Y-%m-%dT%H:%M:%S"]),
+    default=None,
+)
+@click.option(
+    "--date-end",
+    type=click.DateTime(formats=["%Y-%m-%d", "%Y-%m-%dT%H:%M:%S"]),
+    default=None,
+)
+def check_workflows(  # noqa: C901
+    date_start: Optional[datetime.datetime], date_end: Optional[datetime.datetime]
+) -> None:
+    """Check if selected workflows are in sync with database according to predefined rules."""
+    from reana_commons.k8s.api_client import current_k8s_corev1_api_client
+    from reana_commons.config import REANA_RUNTIME_KUBERNETES_NAMESPACE
+    from reana_server.reana_admin.consumer import CollectingConsumer
+    from kubernetes.client.rest import ApiException
+
+    # filter workflows that need to be checked
+    statuses_to_check = [
+        RunStatus.queued,
+        RunStatus.pending,
+        RunStatus.running,
+        RunStatus.finished,
+        RunStatus.failed,
+    ]
+
+    query_filter = [Workflow.status.in_(statuses_to_check)]
+
+    if date_start:
+        query_filter.append(Workflow.created >= date_start)
+
+    if date_end:
+        query_filter.append(Workflow.created < date_end)
+
+    filtered_workflows = Session.query(Workflow).filter(*query_filter)
+
+    if filtered_workflows.count() == 0:
+        click.secho(
+            "No workflows found. Please, adjust your filter conditions.",
+            fg="red",
+        )
+        return
+
+    click.secho(
+        f"Found {filtered_workflows.count()} workflows. Starting to check them...",
+    )
```

Comment:
*suggestion:* Maybe before proceeding we should ask users whatever they want to analyze workflows? Imagine having 100+ workflows. It might take a while (or not).